flink-table/flink-table-runtime-blink/src/test/java/org/apache/flink/table/runtime/typeutils/BigDecimalTypeInfoTest.java

```diff
@@ -36,16 +33,4 @@
 				new BigDecimalTypeInfo(25, 21)
 		};
 	}
-
-	@Test
-	public void testEquality() {
```

Comment:
Why did you remove this test? Was it because you think it is covered in the `TypeInformationTestBase`?

I had a look into the `TypeInformationTestBase` and I think it tests a weird contract of the `TypeInformation#equals`.
If I am not mistaken most of the `TypeInformation(s)` should not pass this test.

```
// test equality
if (typeInfo == otherTypeInfo) {
	assertTrue("hashCode() returns inconsistent results.", typeInfo.hashCode() == otherTypeInfo.hashCode());
	assertEquals("equals() is false for same object.", typeInfo, otherTypeInfo);
}
// test inequality
else {
	assertNotEquals("equals() returned true for different objects.", typeInfo, otherTypeInfo);
}
```
As I understand it, objects can be equal only if it is the same instance. It is false for the `BigDecimalTypeInfo` and probably for multiple others. You can check that if you change the `getTestData` to:

```
@Override
protected BigDecimalTypeInfo[] getTestData() {
	return new BigDecimalTypeInfo[] {
			new BigDecimalTypeInfo(38, 18),
			new BigDecimalTypeInfo(38, 18),
			new BigDecimalTypeInfo(17, 0),
			new BigDecimalTypeInfo(25, 21)
	};
}
```

Reply:
After an offline discussion, I understood it is actually not the problem. This check is actually a precondition for the data. The `equals` is covered in the `TypeInformationTestBase#testSerialization` thus a separate test can be removed.

Resolving this comment.